bokulich-lab/q2-fondue
# ENH: add an action to search for run IDs using a text query

This PR adds a new action allowing retrieval of SRA run IDs using a text search query which will be executed on the BioSample database. The output of the action will then be a single `NCBIAccessionIDs` artifact that can be used to fetch corresponding sequences and/or metadata.

Additionally, a little refactor is included in the Entrezpy pipelines submodule: the ESearch part is separated from the rest of the pipeline to allow fetching more than 10000 UIDs (particularly important for the action introduced in this PR) - see the expanded comments in the code.

**Testing:**
To test, you can run something like:
```
qiime fondue get-ids-from-query --p-email <your email> --p-n-jobs 4 --p-query "txid410656[Organism] AND \"public\"[Filter] AND (chicken OR poultry)" --o-ids ids-from-query.qza --verbose 
```
Please also verify that fetching metadata using run and aggregate IDs works correctly as these parts are influenced by changes in this PR (so, ideally: run, BioProject _and_ study/experiment/sample IDs). 

**Note:**
This PR should be merged after #138 - it already has its changes incorporated (otherwise the CI is broken). Also, it would be easier to review after the other one.
